test_aodncore/pipeline/steps/test_check.py

```diff
@@ -223,3 +235,16 @@ def test_missing_schema_file(self):
         self.assertFalse(check_result.compliant)
         self.assertFalse(check_result.errors)
         self.assertNotEqual(check_result.log, [])
+
+    def test_pattern_schema_match(self):
+        ts_runner = TableSchemaCheckRunner(dummy_config(), self.test_logger,
+                                           check_params={"tableschema_filename_pattern": "[^_]*(_[^_]*)?"}
+                                           )
+        collection = PipelineFileCollection(SAME_SCHEMA_CSVS)
+        ts_runner.run(collection)
+
+        failed_files = [(f.name, f.check_result.log)
+                        for f in collection
+                        if not f.check_result.compliant
+                        ]
+        self.assertEqual([], failed_files)
```

Comment:
You could do a length check, instead of comparing two empty tuples.

Reply:
:+1: Actually I didn't even need to create a list (it was just helpful while debugging)